PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightsScalaProjectWizard.java

```diff
@@ -19,71 +19,66 @@
 import org.eclipse.jdt.ui.wizards.NewJavaProjectWizardPageTwo;
-import org.eclipse.jface.dialogs.IDialogConstants;
 import org.eclipse.swt.SWT;
-import org.eclipse.swt.layout.GridData;
-import org.eclipse.swt.layout.GridLayout;
 import org.eclipse.swt.widgets.Composite;
-import org.eclipse.swt.widgets.Control;
-import org.eclipse.swt.widgets.Display;
-import org.eclipse.swt.widgets.Group;
-import org.scalaide.core.SdtConstants;
-import org.scalaide.ui.ScalaImages;
-import org.scalaide.ui.internal.wizards.NewScalaProjectWizardPageOne;
+import org.eclipse.swt.widgets.Shell;
 
-import com.microsoft.azuretools.adauth.StringUtils;
+import com.microsoft.azuretools.core.utils.PluginUtil;
 import com.microsoft.azuretools.hdinsight.Activator;
-import com.microsoft.tooling.msservices.components.DefaultLoader;
 
 public class HDInsightsScalaProjectWizard extends JavaProjectWizard implements IExecutableExtension {
 	private String id;
 	private Composite sparkLibraryOptionsPanel;
 	private HDInsightScalaPageOne pageOne;
 	private HDInsightScalaPageTwo pageTwo;
-	private static boolean canFinish = false;
+	public static boolean canFinish = false;
 	
 	public HDInsightsScalaProjectWizard() {
-		this(new HDInsightScalaPageOne());
+		this(
+				PluginUtil.forceInstallPluginUsingMarketplace(
+						PluginUtil.scalaPluginSymbolicName, 
+						PluginUtil.scalaPluginMarketplaceURL),
+				setFocusToInstallationWindow(),
+				new HDInsightScalaPageOne()
+				);
 	}
 
-	public HDInsightsScalaProjectWizard(HDInsightScalaPageOne page1) {
-		this(page1, new HDInsightScalaPageTwo(page1));
+	public HDInsightsScalaProjectWizard(boolean checkScalaPlugin, boolean setFocusToInstallationWindow, HDInsightScalaPageOne hdInsightScalaPageOne) {
+		this(hdInsightScalaPageOne, new HDInsightScalaPageTwo(hdInsightScalaPageOne));
 	}
 
 	public HDInsightsScalaProjectWizard(HDInsightScalaPageOne page1, HDInsightScalaPageTwo page2) {
-		super(page1, page2);
+		super((NewJavaProjectWizardPageOne)page1, (NewJavaProjectWizardPageTwo)page2);
 		canFinish = false;
 		pageOne = page1;
 		pageTwo = page2;
 		setWindowTitle("New HDInsight Scala Project");
-		setDefaultPageImageDescriptor(ScalaImages.SCALA_PROJECT_WIZARD());
-		page1.setTitle("Create a Scala project");
-		page1.setDescription("Create a Scala project in the workspace or in an external location.");
+
 		page2.setTitle("HDInsight Spark Project Library Settings");
 		page2.setDescription("Define the project build settings.");
 	}
+	
+	private static boolean setFocusToInstallationWindow() {
+		// Unfortunately, the marketplace client windows can not get the focus so do the trick here since the parent is the project wizard
+		if (PluginUtil.checkPlugInInstallation(PluginUtil.scalaPluginSymbolicName) != true) {
+			Shell parent = PluginUtil.getParentShell();
+			
+			int isModeless = parent.getStyle() & SWT.MODELESS; 
+			if (isModeless == 0) {
+				parent.close();
+			}
+		}
+		
+		return true;
```

Comment:
Discussed, this is to workaround the constructor issue.

---
In reply to: [132864276](https://github.com/Microsoft/azure-tools-for-java/pull/884#discussion_r132864276) [](ancestors = 132864276)